### DISCUSSION TITLE
Capsule and sphere with same height/radius are positioned differently

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#1022] [Q&A] [answered]
Help me understand how a capsule works -- first image is a capsule object, with a radius of 6 and a height of 16:

![image](https://github.com/jrouwe/JoltPhysics/assets/159929665/04a5b917-c08e-4950-9517-83e908101f03)

Second image is a sphere object, with a radius of 16:

![image](https://github.com/jrouwe/JoltPhysics/assets/159929665/bd5d5bea-92b0-4859-892f-9397f77d1198)

The center of these objects should be identical, yet the player object featured here 'floats' above the ground just a bit as a capsule, as opposed to a sphere.

Capsule is created like so:
	JPH::BodyCreationSettings bcs = JPH::BodyCreationSettings(new JPH::CapsuleShape((object->height - object->radius) / 2, object->radius), JPH::Vec3(object->GetPos()->x, object->GetPos()->y, object->GetPos()->z), mat.GetRotation().GetQuaternion(), JPH::EMotionType::Dynamic, Layers::MOVING);

Sphere is created like so:
	JPH::BodyCreationSettings bcs = JPH::BodyCreationSettings(new JPH::SphereShape(object->radius), JPH::Vec3(object->GetPos()->x, object->GetPos()->y, object->GetPos()->z), mat.GetRotation().GetQuaternion(), JPH::EMotionType::Dynamic, Layers::MOVING);

Am I misunderstanding how the size of capsules are measured? It would be cool if there was a diagram in the documentation visualizing how the width/height of a capsule is interpreted.

**Comment by jrouwe** [comment, marked as answer]
You need to subtract 2x the radius to get the height of the cylinder part of the capsule:

`(object->height - object->radius) / 2` -> `(object->height - 2 * object->radius) / 2`

**Reply by SSNTails** [reply]
Hmm, I'm still getting the 'floating' effect:
![image](https://github.com/jrouwe/JoltPhysics/assets/159929665/97bd8813-7a66-4b4c-805d-c71fb1159f76)

JPH::BodyCreationSettings bcs = JPH::BodyCreationSettings(new JPH::CapsuleShape((object->height - object->radius * 2) / 2, object->radius),
	mat.GetTranslation(), mat.GetRotation().GetQuaternion(), JPH::EMotionType::Dynamic, Layers::MOVING);

**Reply by SSNTails** [reply]
I'll hook up the physics drawer next so I can visualize the size of the sphere vs the capsule as a double check.

**Reply by SSNTails** [reply]
Yep, it was a piece of legacy code from the Bullet implementation that was messing up the rendering transform. All good now!
